lib/Service/AvatarService.php

```diff
@@ -0,0 +1,149 @@
+<?php
+
+/**
+ * @author Jakob Sack <mail@jakobsack.de>
+ *
+ * Mail
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OCA\Mail\Service;
+
+use Exception;
+use OCA\Mail\Service\ContactsIntegration;
+use OCP\IUserManager;
+use OCP\IAvatarManager;
+use OCP\IURLGenerator;
+use OCP\Http\Client\IClientService;
+use Gravatar\Gravatar;
+use Mpclarkson\IconScraper\Scraper;
+
+class AvatarService {
```

Comment:
I'd really like to have this class tested. In its current form it's a bit difficult to do so. May I suggest to refactor this class into three classes for the individual avatar sources and just combine their results in this class? I can also take care of this, just let me know :)